Merge with upstream (fix build errors with libavcodec version 3)

meyerj opened the pull request:
I needed https://github.com/RobotWebTools/web_video_server/pull/43 to compile with a newer version of libav/ffmpeg installed on my system, which will be the default starting from Ubuntu zesty. Instead of picking only this PR I merged the latest development branch from upstream and resolved the merge conflicts.

A new streamer type `RosCompressedStreamer` has been added in https://github.com/RobotWebTools/web_video_server/pull/16, which passes through compressed image data from ROS. I adapted the minimum rate publishing from the `ImageStreamer::restreamFrame(double max_age)` implementation.

Other patches also seem to be reasonable or even critical, e.g. https://github.com/RobotWebTools/web_video_server/pull/53.

This branch is completely untested. Carefully review if needed, or simply ignore.

Commit:
added verbose flag

Commit:
changelog updated

Commit:
0.0.3

Commit:
Added default transport parameter for regular image streamers

Commit:
Switched from passing image transport to passing node handle to streamer constructors

Commit:
Added ros compressed video streamer type
This directly passes the ros compressed frame data to the http socket without reencoding it

Commit:
Upgrade for change in signature of async_web_server_cpp request handler

Commit:
Switch to checkout async_web_server_cpp from source

rctoris committed:
Merge pull request #16 from mitchellwills/compressed

Adds support for streaming ROS compressed image topics without the need to reencode them

Commit:
changelog updated

Commit:
0.0.4

BennyRe committed:
More detailed exception message

Programm behavior is not changed since the exception is rethrown.

BennyRe committed:
Removed empty line

rctoris committed:
Merge pull request #23 from iki-wgt/develop

More information when server creation is failed

Commit:
update changelog

Commit:
0.0.5

Commit:
Fixed topic list to display all image topics, fixing Issue #18.

jihoonl committed:
Merge pull request #28 from MasterEric/develop

Fixed topic list to display all image topics, fixing Issue #18.

Commit:
update change log

Commit:
0.0.6

achim-k committed:
web_video_server: fix bool function not returning

This fix is required when compiling the package with `clang`. Otherwise a SIGILL (Illegal instruction) is triggered.

jihoonl committed:
Merge pull request #33 from achim-k/patch-1

web_video_server: fix bool function not returning

artifactz committed:
fixed misuse of remove_if (#35)

jihoonl committed:
Update travis configuration to test against kinetic (#44)

russelhowe committed:
Ffmpeg 3 (#43)

* Correct use of deprecated parameters

codec_context_->rc_buffer_aggressivity marked as "currently useless", so removed
codec_context_->frame_skip_threshold access through new priv_data api

* New names for pixel formats

* AVPicture is deprecated, use AVFrame

* Switch to non-deprecated free functions

* Use new encoding api for newer versions

encode_video2 is deprecated

* codec_context is deprecated, use packet flags

Commit:
update changelog

Commit:
0.0.7

jihoonl committed:
Use trusty instead of xenial.  See travis-ci/travis-ci#7260 (#49)

* Also see RobotWebTools/rosbridge_suite#311

flixr committed:
use SteadyTimer for cleaning up inactive streams (#45)

so that cleanup works correctly even if system time changes

jihoonl committed:
Revert "use SteadyTimer for cleaning up inactive streams (#45)" (#51)

This reverts commit ae74f19ada22f288a7c7a99ada7a1b9b6037c7ce.

randoms committed:
Fix Build for ubuntu 14.04 (#48)

* fix issue #47
* fix double free

randoms committed:
Set image streamer as inactive if topic is not available (#53)

* Resolves #38

jihoonl committed:
Add Indigo test matrix in travis configuration (#50)

randoms committed:
Fix vp8 in kinetic add vp9 and h264 support (#52)

* fix vp8 in kinetic
* add h264 and vp9 support

meyerj committed:
Merge remote-tracking branch 'upstream/develop' into develop-minimum-rate-stream

achim-k commented:
I tested it locally with Pickit indigo-devel and everything seems to work as usual. Actually I don't understand why we do not use the released version and instead use our own fork which is rarely updated. @viktorku do you know more? The commits that I saw on our own fork are not relevant anymore or have been fixed upstream as well.

I tested the upstream version and it also works fine. Will create a PR for 1.9 where we can discuss if we want to use it.

For now I wouldn't merge this PR. If we create a new 1.8 release, we would also include these changes here which I don't feel comfortable with at the moment. I guess we would need to create a `release_1.8` branch in this repo than as well (not sure if rosinstall supports commit hashes).

jabailey committed:
Fix segfault in libav_streamer destructor (resolves #59) (#60)

flixr committed:
use SteadyTimer (if available) for cleaning up inactive streams (#63)

* use SteadyTimer for cleaning up inactive streams

so that cleanup works correctly even if system time changes

SteadyTimer is available since roscpp 1.13.1

* possibility to use SteadyTimer on older ROS versions

when SteadyTimer has been backported to those...

schallerr committed:
Avoid queuing of images on slow ethernet connection (#64)

Commit:
update changelog

Commit:
0.1.0

klintan committed:
fixed undeclared CODEC_FLAG_GLOBAL_HEADER (#65)

dirk-thomas committed:
Pkg format 2 (#68)

* use package format 2

* add missing dependency on sensor_msgs

flixr committed:
fix SteadyTimer check for backported ROS versions (#71)

i.e. on current kinetic

Commit:
Add PngStreamer (#74)

KazutoMurase committed:
lax rule for topic name (#77)

sfalexrog committed:
Add a workaround for MultipartStream constant busy state (#83)

* Add a workaround for MultipartStream constant busy state
* Remove C++11 features

sfalexrog committed:
Add "default_stream_type" parameter (#84)

This allows users to specify default stream type in their .launch files. Using a "ros_compressed" stream type sometimes
results in a much lower resource consumption, and having it set as a default is much nicer for end users.

Commit:
update changelog

Commit:
0.2.0

sfalexrog committed:
Fall back to mjpeg if ros_compressed is unavailable (#87)

meyerj committed:
Merge branch 'develop' into merged/develop-minimum-rate-stream

meyerj commented:
@viktorku Can be closed now that https://github.com/Intermodalics/web_video_server/pull/3 has been merged?

I merged the latest upstream `develop` version and resolved some conflicts due to the change from `ros::Time` to `ros::WallTime` in https://github.com/Intermodalics/web_video_server/commit/083332e444b495b7a5cae2079e26819046dc270a.

Our [develop-minimum-rate-stream](https://github.com/Intermodalics/web_video_server/tree/develop-minimum-rate-stream) branch has some additional patches (https://github.com/Intermodalics/web_video_server/compare/develop...develop-minimum-rate-stream) which are not in [custom](https://github.com/Intermodalics/web_video_server/tree/custom) (https://github.com/Intermodalics/web_video_server/compare/develop...custom) anymore.

viktorku commented:
Can be closed. The relevant feature for minimum rate streaming is now (ready for merge) upstream.
